napari/plugins/_builtins.py

```diff
@@ -57,7 +58,10 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
     """
     if isinstance(path, str) and path.endswith('.csv'):
         return csv_reader_function
-    return image_reader_to_layerdata_reader(magic_imread)
+
+    _p = [path] if isinstance(path, str) else path
+    if all(str(x).lower().endswith(tuple(READER_EXTENSIONS)) for x in _p):
+        return image_reader_to_layerdata_reader(magic_imread)
```

Comment:
We _could_ special case zarr to make sure that
`any(p.endswith('.zarr') for p in Path(stuff).absolute().parts)`.